game/scripts/vscripts/Blockade.ts

```diff
@@ -0,0 +1,81 @@
+import "./modifiers/modifier_blocker"
+
+/**
+ * Line-segment blockade with particles.
+ */
+export class Blockade {
+    private static readonly blockerUnitName = "npc_dota_tutorial_blocker"
+    private static readonly blockerModifierName = "modifier_blocker"
+    private static readonly blockerSpacing = 128 // can use "dota_unit_show_collision_radius 1" in console to visualize this
+    private static readonly blockerParticleName = "particles/tree_barrier.vpcf"
+
+    private numBlockers: number
+
+    private blockers: CDOTA_BaseNPC[] | undefined = undefined
+    private particle: ParticleID | undefined = undefined
+
+    /**
+     * Creates a line-segment blockade.
+     * @param startLocation Start location of the blockade.
+     * @param endLocation End location of the blockade.
+     * @returns Dummy-units contained in the blockade.
+     */
+    constructor(public readonly startLocation: Vector, public readonly endLocation: Vector) {
+        // Spawn as many blockers as we need to cover the line-segment using the above spacing.
+        const distance = this.endLocation.__sub(this.startLocation).Length2D()
+        this.numBlockers = Math.ceil(distance / Blockade.blockerSpacing) ?? 1
```

Comment:
Don't think this `?? 1` will ever do anything, what is the point?

Reply:
if distance is 0 (ie. same start and end), I still wanted to spawn one blocker at that location